R/webApiDataSources.R

```diff
@@ -0,0 +1,118 @@
+# @file dataSources
+#
+# Copyright 2019 Observational Health Data Sciences and Informatics
+#
+# This file is part of StudyManagement
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#' Get the results from individual Achilles reports by data Source definition expression
+#'
+#' @details
+#' Obtain the JSON expression from WebAPI for a given dataSource
+#'
+#' @param baseUrl        The base URL for the WebApi instance, for example:
+#'                       "http://server.org:80/WebAPI".
+#' @param sourceKey      The source key for a CDM instance in WebAPI, as defined in the
+#'                       Configuration page
+#' @param reportType     The precomputed report type in Atlas (Achilles report e.g. dashboard, datadensity, person etc)
+#' @return
+#' A list of object with pre computed Achilles results for the sourceKey and Achilles report
+#'
+#' @examples
+#' \dontrun{
+#' # This will obtain pre-computed summary regarding dataSource
+#'
+#' getDataSourceReportForSourceKey(baseUrl = "http://server.org:80/WebAPI", dataSource = 'dataSource', reportType = 'dashboard')
+#' }
+#'
+#' @export
+
+getDataSourceReportForSourceKey <- function(baseUrl,
+                                            sourceKey,
+                                            reportType){
+  url <- sprintf("%s/cdmresults/%s/%s", baseUrl, sourceKey, reportType)
+  .getApiResponseParse(url)
+}
+
+#' Get the all the results from all Achilles reports for all provided data Sources
+#'
+#' @details
+#' Obtains the JSON expression from WebAPI for a list of dataSource, derived from given list of CDMschemas and returns list of tibble dataframes
+#'
+#' @param baseUrl        The base URL for the WebApi instance, for example:
+#'                       "http://server.org:80/WebAPI".
+#' @param sourceKeys    A list of source keys for a CDM instance in WebAPI, as defined in the
+#'                       Configuration page.
+#' @return
+#' A list of object representing pre computed Achilles results for the list of sourceKey's
+#'
+#' @examples
+#' \dontrun{
+#' # This will obtain pre-computed summary regarding dataSource
+#'
+#' getDataSource(baseUrl = "http://server.org:80/WebAPI", cdmDatabaseSchemas = c('schema1','schema2')
+#' }
+#'
+#' @export
+
+getDataSource <- function(baseUrl,sourceKeys){
```

Comment:
Again, very vague function name. Why not `getAchillesData` or similar?

Reply:
'getDataSourceReport'? Is there another way to identify a data source (not using the source key)? Why is it important to use the source key anyway? Wouldn't it make more sense to use the other function (getSourceKeyConfiguration) to find the data source currently linked to the source key, and then use this function to get the data source report for that data source?

Reply:
The API calls for data-source reports (achilles reports) are dependent on sourceKey (sourceKey is used as the alias). The call is of the form.
`url <- sprintf("%s/cdmresults/%s/%s", baseUrl, sourceKey, reportType)`

So, to be unambiguous in naming and to make the API call we have to provide baseUrl + sourceKey + reportType - we have to include sourceKey.